### PR TITLE
doc/radosgw: edit "Overview" in s3select.rst

### PR DESCRIPTION
**Pull request by zdover23**
Edit the "Overview" section in doc/radosgw/s3select.rst.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/radosgw/s3select.rst [hunk 1]
```diff
@@ -7,23 +7,27 @@
...
-filtering and extracting data.  By "pushing down" the query into radosgw, it's
-possible to save a lot of network and CPU(serialization / deserialization).
-
-    **The bigger the object, and the more accurate the query, the better the
-    performance**.
+The **s3 select** engine creates an efficient pipe between user client and
+storage nodes. The s3 select engine works better closer to storage, and worse
+farther from storage. 
+
+The s3select engine makes it possible to use an SQL-like syntax to select a
```
Comment (anthonyeleven) [inline doc/radosgw/s3select.rst:14]: Above we have "s3 select" with a space.  I suggest converging on _S3 Select_

**Review by anthonyeleven — APPROVED** [review]

File: doc/radosgw/s3select.rst [hunk 2]
```diff
@@ -7,23 +7,27 @@
...
+restricted subset of (structured) data stored in an S3 object. The s3select
+engine facilitates the use of higher level, analytic applications (for example:
+SPARK-SQL). The ability of the s3select engine to target a proper subset of
+structed data within an s3 object decreases latency and increases throughput.
```
[the remark this answers is not in the document]
Comment (galsalomon66) [reply doc/radosgw/s3select.rst:18]: `restricted subset of (structured) data stored in an S3 object.`
it is misleading, the CSV is unstructured, JSON is semi-structured, and Parquet is structured. 
maybe it is better to remove the `(structured)`

[the remark this answers is not in the document]
Comment (zdover23) [reply doc/radosgw/s3select.rst:18]: @galsalomon66 I will update this in about seven hours, when I wake up. Thank you for your technical expertise.

File: doc/radosgw/s3select.rst [hunk 3]
```diff
@@ -7,23 +7,27 @@
...
-single column filtered by another column.  As the following query: ``select
-customer-id from s3Object where age>30 and age<65;``
-
-Currently the whole s3-object must be retrieved from OSD via RGW before
-filtering and extracting data.  By "pushing down" the query into radosgw, it's
-possible to save a lot of network and CPU(serialization / deserialization).
-
-    **The bigger the object, and the more accurate the query, the better the
-    performance**.
+The **s3 select** engine creates an efficient pipe between user client and
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/radosgw/s3select.rst:10]: Accepted.

File: doc/radosgw/s3select.rst (hunk 1 above)
Comment (zdover23) [reply doc/radosgw/s3select.rst:14]: I agree with you. Accepted.

File: doc/radosgw/s3select.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/radosgw/s3select.rst:18]: I've removed that parenthetical.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/52219 - Reef backport
https://github.com/ceph/ceph/pull/52220 - Quincy backport

**Pull request merged**